File: app/types/animal.ts
export class Animal {
  constructor(
    public id: number | null,
    public nameAnimal: string,
    public especie: string,
    public raca: string,
    public urlFoto: string,
    public statusAnimal: string,
  ) {}
}

export interface AnimalContextType {
  favoritos: Animal[];
  adicionarFavorito: (animal: Animal) => void;
  removerFavorito: (id: number) => void;
  isFavorito: (id: number) => boolean;
}

export interface AnimalFormProps {
  animalExistente?: Animal;
}

File: backend/src/main/java/com/senac/backend/backend/domain/entities/Animal.java
package com.senac.backend.backend.domain.entities;

import com.senac.backend.backend.domain.enuns.EnumEspecieAnimal;
import com.senac.backend.backend.domain.enuns.EnumStatusAnimal;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table (name = "animal")
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Animal {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String nameAnimal;

    private String raca;

    @Column(columnDefinition = "TEXT")
    private String urlFoto;

    private EnumEspecieAnimal especie;

    private EnumStatusAnimal statusAnimal = EnumStatusAnimal.DISPONIVEL;

}
